packages/flutter/lib/src/widgets/router.dart

```diff
@@ -1052,6 +1052,7 @@ class _BackButtonListenerState extends State<BackButtonListener> {
     if (oldWidget.onBackButtonPressed != widget.onBackButtonPressed) {
       dispatcher?.removeCallback(oldWidget.onBackButtonPressed);
```

Comment:
`ChildBackButtonDispatcher.removeCallback` will remove itself from parent.
```dart
  @override
  void removeCallback(ValueGetter<Future<bool>> callback) {
    super.removeCallback(callback);
    if (!hasCallbacks)
      parent.forget(this);
  }
```

Reply:
Ah nice catch!

Reply:
maybe add `updateCallback` instead?

Reply:
update callback need to remove the old callback, this will not work. The other way to avoid the take priority is to swap the order
```dart
dispatcher?.addCallback(widget.onBackButtonPressed);
dispatcher?.removeCallback(oldWidget.onBackButtonPressed);
```